Sources/Class-Data_Validator.php

```diff
@@ -0,0 +1,1326 @@
+	 * @param string $field
+	 * @param mixed[] $input
+	 * @param mixed[]|null $validation_parameters array or null
+	 */
+	protected function _validate_valid_ipv6($field, $input, $validation_parameters = null)
+	{
+		if (!isset($input[$field]))
+			return;
+
+		if (filter_var($input[$field], FILTER_VALIDATE_IP, FILTER_FLAG_IPV6) === false)
+		{
+			return array(
+				'field' => $field,
+				'input' => $input[$field],
+				'function' => __FUNCTION__,
+				'param' => $validation_parameters
+			);
+		}
+	}
+
+	/**
+	 * valid_ip ... Determine if the provided value is a valid IP4 address
+	 *
+	 * Usage: '[key]' => 'valid_ip'
+	 *
+	 * @param string $field
+	 * @param mixed[] $input
+	 * @param mixed[]|null $validation_parameters array or null
+	 */
+	protected function _validate_valid_ip($field, $input, $validation_parameters = null)
+	{
+		if (!isset($input[$field]))
+			return;
+
+		if (filter_var($input[$field], FILTER_VALIDATE_IP, FILTER_FLAG_IPV4) === false)
+		{
+			return array(
+				'field' => $field,
+				'input' => $input[$field],
+				'function' => __FUNCTION__,
+				'param' => $validation_parameters
+			);
+		}
+	}
+
+	/**
+	 * Validate PHP syntax of an input.
+	 *
+	 * This approach to validation has been inspired by Compuart.
+	 *
+	 * Usage: '[key]' => 'php_syntax'
+	 *
+	 * @uses ParseError
+	 * @param string $field
+	 * @param mixed[] $input
+	 * @param mixed[]|null $validation_parameters array or null
+	 */
+	protected function _validate_php_syntax($field, $input, $validation_parameters = null)
+	{
+		if (!isset($input[$field]))
+			return;
+
+		// Check the depth.
+		$level = 0;
+		$tokens = @token_get_all($input[$field]);
+		foreach ($tokens as $token)
+		{
+			if ($token === '{')
+				$level++;
+			elseif ($token === '}')
+				$level--;
+		}
+
+		if (!empty($level))
+			$result = false;
+		else
+		{
+			// Check the validity of the syntax.
+			ob_start();
+			$errorReporting = error_reporting(0);
+			try
+			{
+				$result = @eval('
+					if (false)
+					{
+						' . preg_replace('~^(?:\s*<\\?(?:php)?|\\?>\s*$)~u', '', $input[$field]) . '
+					}
+				');
+			}
+			catch (ParseError $e)
+			{
+				$result = false;
+			}
+			error_reporting($errorReporting);
+			@ob_end_clean();
+		}
+
+		if ($result === false)
+		{
+			$errorMsg = error_get_last();
+
+			return array(
+				'field' => $field,
+				'input' => $input[$field],
+				'error' => '_validate_php_syntax',
+				'error_msg' => $errorMsg['message'],
+				'param' => $validation_parameters
+			);
+		}
+	}
+
+	/**
+	 * Checks if the input is a valid css-like color
+	 *
+	 * Usage: '[key]' => 'valid_color'
+	 *
+	 * @param string $field
+	 * @param mixed[] $input
+	 * @param mixed[]|null $validation_parameters array or null
+	 */
+	protected function _validate_valid_color($field, $input, $validation_parameters = null)
+	{
+		if (!isset($input[$field]))
+			return;
+
+		// A color can be a name: there are 140 valid, but a similar list is too long, so let's just use the basic 17
```

Comment:
Why not use all of them? The full list is available at https://www.w3schools.com/cssref/css_colors.asp. It should just be a matter of copying and pasting them in.